src/index.ts

```diff
@@ -68,11 +69,43 @@ export async function cli(args: string[]) {
     process.env.NODE_ENV = process.env.NODE_ENV || 'development';
   }
 
+  const config = loadAndValidateConfig(cliFlags, pkgManifest);
+
+  const tsConfig = tsConfigPaths.loadConfig();
+  const matchTsConfigPath =
+    tsConfig.resultType === 'success'
+      ? tsConfigPaths.createMatchPath(tsConfig.absoluteBaseUrl, tsConfig.paths)
+      : () => undefined;
+
+  // Ex: src/foo/bar => /_dist_/foo/bar
+  const expandBareImport = (spec) => {
```

Comment:
instead of passing this actual function in as a top-level option, can it instead be a factory that creates an expander from a config object? Something like:

```js
// Inside of dev, build, scan-imports, etc.
const expandBareImport = createBareImportHandler(config);
```

Reply:
I remember now why I did it this way. So https://github.com/stellarhoof/snowpack/blob/b66f2efcfd7a9a7148187e86306df3f79cca1121/src/index.ts#L77 is relatively expensive to initialize because of fs access and json parsing etc... so we shouldn't do it for every call to `expandBareImport`. This is why `expandBareImport` closes over it. What you suggest would require putting https://github.com/stellarhoof/snowpack/blob/b66f2efcfd7a9a7148187e86306df3f79cca1121/src/index.ts#L75 in some global scope, and the only one I see is `commandOptions`. But at that point it'd be better to just include `expandBareImports` itself. Thoughts?

Reply:
That concern definitely makes sense, but shouldn't be blocking the move to a  `createBareImportHandler` factory. Something like:

```
const tsConfig = ...
const matchTsConfigPath = ...
function createBareImportHandler(config) {
  // Reference the already initialized, global tsConfig & matchTsConfigPath
}
```

Reply:
Done